sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,368 @@
+from metapool import KLSampleSheet, validate_and_scrub_sample_sheet
+from os import stat, listdir, makedirs
+from os.path import join, basename
+from sequence_processing_pipeline.Job import Job
+from sequence_processing_pipeline.PipelineError import PipelineError
+from sequence_processing_pipeline.Pipeline import Pipeline
+from shutil import move
+import logging
+from json import dumps
+from datetime import date
+from sequence_processing_pipeline.Commands import split_similar_size_bins
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+'''
+TODO:
+The only two methods not called in testing are:
+in _filter
+in run
+'''
+
+
+class NuQCJob(Job):
+    def __init__(self, fastq_root_dir, output_path, sample_sheet_path,
+                 minimap_database_paths, queue_name,
+                 node_count, nprocs, wall_time_limit, jmem, fastp_path,
+                 minimap2_path, samtools_path, modules_to_load, qiita_job_id,
+                 pool_size, max_array_length):
+        """
+        Submit a Torque job where the contents of fastq_root_dir are processed
+        using fastp, minimap, and samtools. Human-genome sequences will be
+        filtered out, if needed.
+        :param fastq_root_dir: Path to a dir of Fastq files, org. by project.
+        :param output_path: Path where all pipeline-generated files live.
+        :param sample_sheet_path: Path to a sample sheet file.
+        :param minimap_database_paths: Path to human genome databases in env.
+        :param queue_name: Torque queue name to use in running env.
+        :param node_count: Number of nodes to use in running env.
+        :param nprocs: Number of processes to use in runing env.
+        :param wall_time_limit: Hard wall-clock-time limit (in min) for procs.
+        :param jmem: String representing total memory limit for entire job.
+        :param fastp_path: The path to the fastp executable
+        :param minimap2_path: The path to the minimap2 executable
+        :param samtools_path: The path to the samtools executable
+        :param modules_to_load: A list of Linux module names to load
+        :param qiita_job_id: identify Torque jobs using qiita_job_id
+        :param pool_size: The number of jobs to process concurrently.
+
+        """
+        super().__init__(fastq_root_dir,
+                         output_path,
+                         'QCJob',
+                         [fastp_path, minimap2_path, samtools_path],
+                         max_array_length,
+                         modules_to_load=modules_to_load)
+        self.sample_sheet_path = sample_sheet_path
+        self._file_check(self.sample_sheet_path)
+        metadata = self._process_sample_sheet()
+        self.sample_ids = metadata['sample_ids']
+        self.project_data = metadata['projects']
+        self.needs_trimming = metadata['needs_adapter_trimming']
+        self.nprocs = 16 if nprocs > 16 else nprocs
```

Comment:
why?

Reply:
I believe it's a sanity check from the original shell scripts. @antgonza even after the port to SLURM, I don't think we've needed to go above 16 processes. Should we hold onto this hard upper-limit?

Reply:
nprocs tentatively removed.